TrafficCapture/dockerSolution/src/main/docker/migrationConsole/console_api/console_api/apps/orchestrator/views.py

```diff
@@ -37,70 +36,77 @@ def pretty_request(request, data):
+    pipeline_manager_role_arn = request_data.get('PipelineManagerAssumeRoleArn')
+    if pipeline_manager_role_arn:
+        logger.debug(f'Assuming provided role: {pipeline_manager_role_arn}')
+        session = get_assume_role_session(role_arn=pipeline_manager_role_arn,
+                                          session_name=f'Console{action_name}PipelineAssumeRole')
+        osi_client = session.client('osis')
+    else:
+        osi_client = boto3.client('osis')
+    return osi_client
+
+
+def osi_update_workflow(request, osi_action_func: Callable, action_name: str):
     request_data = request.data
     logger.info(pretty_request(request, request_data))
 
-    osi_serializer = OpenSearchIngestionCreateRequestSerializer(data=request_data)
+    osi_serializer = OpenSearchIngestionUpdateRequestSerializer(data=request_data)
     osi_serializer.is_valid(raise_exception=True)
+    pipeline_name = request_data.get('PipelineName')
     try:
-        osi_client = boto3.client('osis')
-        create_pipeline_from_json(osi_client=osi_client,
-                                  input_json=request_data,
-                                  pipeline_template_path=PIPELINE_TEMPLATE_PATH)
-    except InvalidAuthParameters as i:
-        logger.error(f"Error performing osi_create_migration API: {i}")
-        return Response(str(i), status=status.HTTP_400_BAD_REQUEST)
+        osi_action_func(osi_client=determine_osi_client(request_data=request_data, action_name=action_name),
+                        pipeline_name=pipeline_name)
     except Exception as e:
-        logger.error(f"Error performing osi_create_migration API: {e}")
+        logger.error(f'Error performing OSI {action_name} Pipeline API: {e}')
         return Response(str(e), status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
     response_data = {
-        'Timestamp': datetime.datetime.now(datetime.timezone.utc),
-        'MigrationType': MigrationType.OSI_HISTORICAL_MIGRATION
+        'Timestamp': datetime.datetime.now(datetime.timezone.utc)
     }
     return Response(response_data, status=status.HTTP_200_OK)
 
 
 @api_view(['POST'])
 @parser_classes([JSONParser])
-def osi_start_migration(request):
+def osi_create_migration(request):
     request_data = request.data
     logger.info(pretty_request(request, request_data))
+    action_name = 'Create'
 
-    pipeline_name = request_data.get("PipelineName")
+    osi_serializer = OpenSearchIngestionCreateRequestSerializer(data=request_data)
+    osi_serializer.is_valid(raise_exception=True)
     try:
-        osi_client = boto3.client('osis')
-        start_pipeline(osi_client=osi_client, pipeline_name=pipeline_name)
+        create_pipeline_from_json(osi_client=determine_osi_client(request_data=request_data, action_name=action_name),
+                                  input_json=request_data,
+                                  pipeline_template_path=PIPELINE_TEMPLATE_PATH)
+    except InvalidAuthParameters as i:
```

Comment:
Minor note that at some point, it would be great to have a test that exercised this branch, but not a blocker for this PR